contents/data_vis_makie_recipe.md

```diff
@@ -0,0 +1,96 @@
+## A Makie recipe for a DataFrame {#sec:recipe_df}
+
+Unlike other libraries that already support a wide set of input formats via recipes, i.e. `Plots.jl`, in `Makie.jl` most of the time we need to pass the raw data to our functions.
+However, if we want to define our own `recipe` in Makie we could also do it.
+As an example we will code a small **full recipe** for a `DataFrame`. Please refer to the [documentation](https://makie.juliaplots.org/stable/documentation/recipes/) for more details.
+
+A Makie `recipe` consist of two parts, a plot `type` name defined via `@recipe` and a custom `plot!`(`Makie.plot!`) which creates the actual plot via plotting functions already defined.
+
+```jl
+s = """
+    @recipe(DfPlot, df) do scene
+        Attributes(
+            x = :A,
+            y = :B,
+            c = :C,
+            color = :red,
+            colormap = :plasma,
+            markersize = 20,
+            marker = :rect
+        )
+    end
+    """
+sc(s)
+```
+
+with this we will have `dfplot` and `dfplot!` conveniently defined for us.
+Now, let's take care of our plot definition. We will do a simple scatter plot.
+
+```
+import Makie
+```
+
+```jl
+s = """
+    function Makie.plot!(p::DfPlot{<:Tuple{<:DataFrame}})
+        df = p[:df][]
+        x = getproperty(df, p[:x][])
+        y = getproperty(df, p[:y][])
+        c = getproperty(df, p[:c][])
+        scatter!(p, x, y; color = c, markersize = p[:markersize][],
+            colormap = p[:colormap][], marker = p[:marker][])
+        return p
+    end
+    """
+sc(s)
+```
+
+Note the extras `[]` at the end of each variable. Those are due to the fact that *recipes* in Makie are dynamic, meaning that our plots will update if our variables change. See [observables](https://makie.juliaplots.org/stable/documentation/nodes/) to know more.
```

Comment:
Ah, is it because the DfPlot is immutable and they use pointers to (de)reference stuff? Sounds a bit dangerous but okay

Reply:
is the whole logic of observables in Makie 😄